recipes/consensusfixer/ConsensusFixer.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+# ConsensusFixer wrapper script
+# based on picard wrapper
+set -eu -o pipefail
+
+set -o pipefail
+export LC_ALL=en_US.UTF-8
+
+# Find original directory of bash script, resovling symlinks
+# http://stackoverflow.com/questions/59895/can-a-bash-script-tell-what-directory-its-stored-in/246128#246128
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+    DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+    SOURCE="$(readlink "$SOURCE")"
+    [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+
+JAR_DIR=$DIR
+ENV_PREFIX="$(dirname $(dirname $DIR))"
+# Use Java installed with Anaconda to ensure correct version
+java="$ENV_PREFIX/bin/java"
+
+# if JAVA_HOME is set (non-empty), use it. Otherwise keep "java"
+if [ -z "${JAVA_HOME:=}" ]; then
+  if [ -e "$JAVA_HOME/bin/java" ]; then
+      java="$JAVA_HOME/bin/java"
+  fi
+fi
+
+# extract memory and system property Java arguments from the list of provided arguments
+# http://java.dzone.com/articles/better-java-shell-script
+default_jvm_mem_opts="-Xms512m -Xmx1G"
+jvm_mem_opts=""
+jvm_prop_opts=""
+pass_args=""
+for arg in "$@"; do
+    case $arg in
+        '-D'*)
+            jvm_prop_opts="$jvm_prop_opts $arg"
+            ;;
+        '-XX'*)
+            jvm_prop_opts="$jvm_prop_opts $arg"
+            ;;
+         '-Xm'*)
+            jvm_mem_opts="$jvm_mem_opts $arg"
+            ;;
+         *)
+            pass_args="$pass_args $arg"
+            ;;
+    esac
+done
+
+if [ "$jvm_mem_opts" == "" ]; then
+    jvm_mem_opts="$default_jvm_mem_opts"
+fi
+
+pass_arr=($pass_args)
+if [[ ${pass_arr[0]:=} == org* ]]
+then
+    eval "$java" $jvm_mem_opts $jvm_prop_opts -cp "$JAR_DIR/ConsensusFixer.jar" $pass_args
```

Comment:
`eval` is usually never necessary in such circumstances, have you tried it without it?

Reply:
`eval` will e.g. re-interpolate any pending `$` in the strings, etc.

I think, in the mind of the original wrapper's author, `eval` enables to  handle arguments containing white-spaces (that would otherwise be destroyed by the while loop), by wrapping additional quotes around them or giving extra back-slashes. (Similar to one approach of handling remote white-spaces in `rsync`).

(I would have personally went for much simpler handling through bash arrays and quoted array expansion.
But, again, I preferred to reuse whatever is already in bioconda)

Reply:
@DrYak the proper way to handle whitespace in bash is to use arrays. Using the peptide shaker is also an option, although I think this bash script can be made to work fine. Your choice really

Reply:
Choose the *redo it properly using arrays* route.